IPython/core/interactiveshell.py

```diff
@@ -2613,7 +2615,7 @@ def run_cell(self, raw_cell, store_history=False, silent=False, shell_futures=Tr
         result = ExecutionResult()
 
         if (not raw_cell) or raw_cell.isspace():
-            self.last_execution_succeeded = True
+            self.last_execution_succeeded, self.last_execution_result = True, result
```

Comment:
Let's set these on two lines, rather than doing two things on one line. It's a bit longer, but it makes it clear and simple.